tests/fixture/tmpnet/node.go

```
@@ -56,6 +56,27 @@ type NodeRuntimeConfig struct {
 	Kube    *KubeRuntimeConfig    `json:"kube,omitempty"`
 }
 
+// GetNetworkStartTimeout returns the timeout to use when starting a network.
+func (c *NodeRuntimeConfig) GetNetworkStartTimeout(nodeCount int) time.Duration {
+	switch {
+	case c.Process != nil:
+		// Processes are expected to start quickly, nodeCount is ignored
+		return DefaultNetworkTimeout
+	case c.Kube != nil:
+		// Ensure sufficient time for scheduling and image pull
+		timeout := time.Duration(nodeCount) * time.Minute
+
+		if c.Kube.UseExclusiveScheduling {
+			// Ensure sufficient time for the creation of autoscaled nodes
+			timeout *= 2
+		}
+
+		return timeout
+	default:
+		panic("no runtime configuration set")
```

Consumers of this method handle errors properly
- https://github.com/ava-labs/avalanchego/blob/c3ff3cb0ff9d0dc5e2761160f19e7e3a81f638b5/tests/fixture/tmpnet/tmpnetctl/main.go#L83
- https://github.com/ava-labs/avalanchego/blob/c3ff3cb0ff9d0dc5e2761160f19e7e3a81f638b5/tests/fixture/e2e/helpers.go#L285

What's the rationale for panicking instead of returning an error? If panic is required, the method name should clearly indicate it can panic (e.g., `MustDoSomething`).